houndigrade/cli.py

```diff
@@ -0,0 +1,214 @@
+
+    results = {
+        'cloud': cloud,
+        'inspection_targets': target,
+        'facts': {},
+    }
+
+    for image_id, drive in target:
+        click.echo('Checking drive {}'.format(drive))
+
+        results['facts'][drive] = {'image_id': image_id}
+        for partition in get_partitions(drive):
+            click.echo('Checking partition {}'.format(partition))
+
+            results['facts'][drive][partition] = []
+            try:
+                subprocess.run([
+                    'mount',
+                    '-t',
+                    'auto',
+                    '{}'.format(partition),
+                    '{}'.format(INSPECT_PATH)
+                ],
+                    check=True,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    universal_newlines=True
+                )
+            except subprocess.CalledProcessError as e:
+                click.echo(
+                    _('Mount of {} failed '
+                      'with error: {}'.format(partition, e.stderr)),
+                    err=True
+                )
+
+                results['facts'][drive][partition].append({
+                    'error': e.stderr
+                })
+
+                continue
+
+            check_release_files(partition, results['facts'][drive][partition])
+
+            click.echo('Unmounting partition: {}'.format(partition))
+            subprocess.run(['umount', '{}'.format(INSPECT_PATH)])
+
+    if debug:
+        click.echo(json.dumps(results))
+
+    report_results(results)
+
+
+def report_results(results):
+    """
+    Places the results on a queue.
+
+    Args:
+        results (dict): The results of the finished inspection.
+
+    """
+    queue_name = os.getenv('RABBITMQ_QUEUE_NAME')
+
+    exchange = Exchange(os.getenv('RABBITMQ_EXCHANGE_NAME'), durable=True)
+    queue = Queue(queue_name, exchange=exchange, routing_key=queue_name)
+
+    with Connection(os.getenv('RABBITMQ_URL')) as conn:
+        producer = conn.Producer(serializer='json')
+        producer.publish(
+            results,
+            exchange=exchange,
+            routing_key=queue_name,
+            declare=[queue]
+        )
+
+
+def check_release_files(partition, results):
+    """
+    Check os release files to see if they indicate RHEL.
+
+    Args:
+        partition (str): The partition we are currently checking.
+        results (list): Part of the results dict to which we should be
+        writing our results.
+
+    """
+    release_file_paths = find_release_files()
+
+    if not release_file_paths:
+        click.echo('No release files found on {}'.format(partition))
+        results.append({
+            'rhel_found': False,
+            'status': 'No release files found on {}'.format(partition)
+        })
+    else:
+        for release_file_path in release_file_paths:
+            rhel_found, file = check_file(release_file_path)
```

Comment:
Python3 🎉 

That said I'm open to changing it to a different name if you have suggestions.

Reply:
Seconded on the `file` name. Conventions I've seen around this can be as simple as `the_file` or `_file`, but here I think a more descriptive name like `contents` or `file_contents` would be appropriate since it looks like that's what this really is.